examples/relay-server/src/main.rs

```diff
@@ -96,14 +114,57 @@ async fn main() -> Result<(), Box<dyn Error>> {
 
                 println!("{event:?}")
             }
-            SwarmEvent::NewListenAddr { address, .. } => {
+            SwarmEvent::NewListenAddr { mut address, .. } => {
+                address.push(Protocol::P2p(local_peer_id));
                 println!("Listening on {address:?}");
             }
             _ => {}
         }
     }
 }
 
+fn listen_on_websocket(swarm: &mut Swarm<Behaviour>, opt: &Opt) -> Result<(), Box<dyn Error>> {
+    match opt.websocket_port {
+        Some(0) => {
+            return Err(Box::new(std::io::Error::new(std::io::ErrorKind::InvalidInput,
+            "Websocket port is 0, which is not supported in this example, since it will use a non-deterministic port. Please use a fixed port for websocket.")));
+        }
+        Some(port) => {
+            let address = Multiaddr::from(Ipv4Addr::UNSPECIFIED)
+                .with(Protocol::Tcp(port))
+                .with(Protocol::Ws(std::borrow::Cow::Borrowed("/")));
+
+            info!(?address, "Listening on webSocket");
+            swarm.listen_on(address.clone())?;
+        }
+        None => {
+            info!("Does not use websocket");
+        }
+    }
+    Ok(())
+}
+
+fn listen_on_webrtc(swarm: &mut Swarm<Behaviour>, opt: &Opt) -> Result<(), Box<dyn Error>> {
+    match opt.webrtc_port {
+        Some(0) => {
+            return Err(Box::new(std::io::Error::new(std::io::ErrorKind::InvalidInput,
+            "Websocket port is 0, which is not supported in this example, since it will use a non-deterministic port. Please use a fixed port for webRTC.")));
+        }
```

Comment:
Why shouldn't we support this? We also support non-deterministic ports for the basic tcp and quic transports.

Reply:
You right, I concentrate on creating deterministic addresses and did not pay attention for that case. I will fix it.